### PR TITLE
feat(aws-lambda-ssm-string-parameter): New aws-lambda-ssm-string-parameter pattern implementation (#64)

### LINKED ISSUE #64: New pattern: aws-lambda-ssm-parameter
Giving the user an option to choose the permissions, e.g. read (default) or read-write, from SSM parameter store

For example, check [here](https://github.com/awslabs/aws-solutions-constructs/blob/master/source/patterns/%40aws-solutions-constructs/aws-lambda-dynamodb/lib/index.ts#L53)

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by danielmatuki**
closes #64

Issue #, if available: #64

Description of changes:
Implements the AWS Solutions Construct that creates an AWS Lambda function and AWS Systems Manager Parameter Store String parameter with the least privileged permissions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 957eed770d31778853f7118e98fe88f6a7d284e4
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b6218bb7-e655-427e-a058-db1374124cfb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-ssm-string-parameter/test/lambda-ssm-string-parameter.test.ts [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+  // Stack
+  const stack = new Stack();
+
+  const testVpc = new ec2.Vpc(stack, "test-vpc", {});
+
+  const app = () => {
+    // Helper declaration
+    new LambdaToSsmStringParameter(stack, "lambda-to-ssm-stack", {
+      lambdaFunctionProps: {
+        runtime: lambda.Runtime.NODEJS_10_X,
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-ssm-string-parameter/test/lambda-ssm-string-parameter.test.ts:379]: NodeJS version...

File: source/patterns/@aws-solutions-constructs/core/lib/ssm-parameter-helper.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Method to build the default AWS SSM Parameter Store
+ *
+ * @param scope
+ * @param id
+ * @param stringParameterProps
+ */
+export function buildSsmStringParamter(scope: Construct, id: string, stringParameterProps: StringParameterProps): StringParameter {
+  return new StringParameter(scope, id, stringParameterProps);
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/ssm-parameter-helper.ts:25]: So I'm looking at possible defaults, and see that there really aren't any. But I also see parameter Type as a member of StringParameterProps. What happens if you send SECURE_STRING or STRINGLIST? Does it get ignored? The docs are pretty clear that you can't create a SECURE_STRING parameter with the CDK, and StringList has a completely separate  L2 construct.

If it must be STRING, then we should overrule whatever the user sent and dump a warning that we are doing so. Also should point it out in the docs.

File: source/patterns/@aws-solutions-constructs/core/lib/ssm-parameter-helper.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/**
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/ssm-parameter-helper.ts:1]: should be ssm-string-parameter-helper

**Review by biffgaut — COMMENTED** [review]
A few changes, mostly naming and the like.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 4093afecd4211e0997bf780d1ba69e2ecf7c4f78
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=667d3946-2c13-4c2c-87b2-513843b9d362%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: fe86781744da4423481489ba68304fcec9246dfa
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0baf67ce-ea7e-4ea3-81bd-0ff117083e8d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 61bfbafb5ac72ccbb018c3b62d876f2263630de2
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2ff5a371-2762-40c1-8fa6-1b6e01e70aca%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**